### ISSUE TITLE
Visualize incorrectly reports all lines covered

### ISSUE THREAD

**Issue by zhassan-aws** [model-checking/kani#2048]
I tried this code:

```rust
#[kani::proof]
fn main() {
    let x: u8 = kani::any();
    if x > 5 {
        if x < 2 {
            let y = x;
        }
    } else {
        assert!(x < 10);
    }
}
```

using the following command line invocation:

```
kani test.rs --visualize
```

with Kani version: 0.18.0

I expected to see this happen: Coverage is not 100% as line 6 is unreachable due to the contradicting conditions.

Instead, this happened: The report from visualize says all lines are reachable:

![image](https://user-images.githubusercontent.com/88045115/209881705-c7d78b34-3141-47e0-9b94-fd69eb43eb74.png)

Line 6 is displayed in orange (as well as line 5!) rather than red:

![image](https://user-images.githubusercontent.com/88045115/209881772-0525e6bb-6f92-4f21-ab84-8cdde9717c1c.png)

**Comment by tautschnig** [comment]
I am not sure whether this problem actually persists, but the following patch may be important for it fixes location information for function calls:
````patch
--- a/kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs
+++ b/kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs
@@ -564,7 +564,7 @@ fn codegen_funcall(
                     | InstanceDef::CloneShim(..) => {
                         // We need to handle FnDef items in a special way because `codegen_operand` compiles them to dummy structs.
                         // (cf. the function documentation)
-                        let func_exp = self.codegen_func_expr(instance, None);
+                        let func_exp = self.codegen_func_expr(instance, None).with_location(loc);
                         vec![
                             self.codegen_expr_to_place(destination, func_exp.call(fargs))
                                 .with_location(loc),
````

**Comment by celinval** [comment]
Closing this since we are now deprecating `--visualize` option.
